automerge-frontend/src/mutation.rs

```diff
@@ -82,6 +83,36 @@ impl LocalChange {
     }
 }
 
+// A container for either a multivalue or a multigrapheme
+enum MultiThing {
+    Value(MultiValue),
+    Grapheme(MultiGrapheme),
+}
+
+impl MultiThing {
+    fn into_value(self) -> MultiValue {
+        match self {
+            Self::Value(value) => value,
+            Self::Grapheme(_) => unreachable!(),
+        }
+    }
+
+    fn into_grapheme(self) -> MultiGrapheme {
+        match self {
+            Self::Value(_) => unreachable!(),
+            Self::Grapheme(grapheme) => grapheme,
+        }
+    }
+}
+
+enum LocalOperationForRollback {
```

Comment:
We could avoid the `MultiThing` wrapper by making the operations specific to the data type right? i.e `Set` for everything which needs a `MultiValue` and `SetText` for anything which needs a `MultiGrapheme`?

Reply:
Ah yes, that's a much better idea! (I've also added a `SetList` as it expects the old value to always exist unlike in a map so avoids an unwrap).